# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/windows/WinHeaderOnlyUtils.h
# Commit: cae98337ddda
# Full Hash: cae98337ddda1b78db120235072ef5154cdf65f5
# Author: Aaron Klotz <aklotz@mozilla.com>
# Date: 2019-07-24 09:54:28
# Regressor Bug: 1567614
# File Overlap Count: 1
# Description:
#   Bug 1567614: Part 2 - Add ShellExecuteByExplorer overload to handle absolute PIDL lists; r=jmathies
#   
#   For the URI handling case, we still want to parse the URI to look for any
#   malformation. OTOH, IShellDispatch2::ShellExecute does not accept PIDLs as
#   arguments, we we need an overload that converts the absolute PIDL back to a
# ==============================================================================

diff -r e8bf96e8321a -r cae98337ddda widget/windows/WinHeaderOnlyUtils.h
--- a/widget/windows/WinHeaderOnlyUtils.h	Tue Jul 23 20:17:58 2019 +0000
+++ b/widget/windows/WinHeaderOnlyUtils.h	Tue Jul 23 20:18:07 2019 +0000
@@ -11,6 +11,7 @@
 #include <winerror.h>
 #include <winnt.h>
 #include <winternl.h>
+#include <objbase.h>
 
 #include <stdlib.h>
 
@@ -532,6 +533,10 @@
 }
 #endif  // defined(MOZILLA_INTERNAL_API)
 
+struct CoTaskMemFreeDeleter {
+  void operator()(void* aPtr) { ::CoTaskMemFree(aPtr); }
+};
+
 }  // namespace mozilla
 
 #endif  // mozilla_WinHeaderOnlyUtils_h
